File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/helloworld/HelloWorld.java
package de.maibornwolff.domainchoreograph.examples.helloworld;

import de.maibornwolff.domainchoreograph.core.api.DomainDefinition;
import de.maibornwolff.domainchoreograph.core.api.DomainFunction;

@DomainDefinition
public class HelloWorld {

    private String message;

    public HelloWorld(String message) {
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    @DomainFunction
    public static HelloWorld resolveHelloWorld(Name name) {
        return new HelloWorld("Hello " + name.getValue());
    }

    @Override
    public String toString() {
        return message;
    }
}

File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/helloworld/HelloWorld.kt
package de.maibornwolff.domainchoreograph.examples.helloworld

import de.maibornwolff.domainchoreograph.core.api.DomainDefinition
import de.maibornwolff.domainchoreograph.core.api.DomainFunction

@DomainDefinition
data class HelloWorld(val message: String) {

    companion object {

        @DomainFunction
        fun resolveHelloWorld(name: Name): HelloWorld {
            return HelloWorld("Hello " + name.value)
        }
    }
}
